src/plugins/workspace/public/index.ts

```diff
@@ -11,3 +11,4 @@ export function plugin() {
 
 export { WorkspacePluginSetup, WorkspaceCollaborator } from './types';
 export { WorkspaceCollaboratorType } from './services';
+export { isNavGroupInFeatureConfigs } from './utils';
```

Comment:
I’m not sure if the required bundling will guarantee that the workspace plugin is always loaded before the explore plugin. Even if it does, directly importing functions this way can create a tight coupling between the two plugins, which can become a maintenance challenge.

It might be cleaner to expose this utility through the start lifecycle method of the workspace plugin, ensuring a more explicit API contract. This approach aligns better with the intended OpenSearch Dashboards plugin architecture, where plugins communicate through well-defined interfaces.

Additionally, tightly coupling these plugins means that future changes to the workspace plugin would require testing against the explore plugin, potentially complicating development.

Reply:
What do you think about moving `isNavGroupInFeatureConfigs` to `src/core`? I think we can move this function to `src/plugins/workspace/public/utils.ts`. Then the workspace plugin and explore plugin can use the same function from core module.

Reply:
There are some workspace utils functions in `src/core/utils/workspace.ts`, and we already introduced `features` as a property of workspace object in `src/core/types/workspace.ts`. The nav group is a concept of core too. So I think the `isNavGroupInFeatureConfigs` can be moved to the core module.

Reply:
ah i didn't notice it's already part of core. refactored it in d918051 to move to core and remove dependency of `workspace` for `explore`. thanks